### PR TITLE
swatch-1240: Remove tag_profile from capacity api and use swatch-product-config

### PR DESCRIPTION
**Pull request by kflahert**
Jira issue: [SWATCH-1240](https://issues.redhat.com/browse/SWATCH-1240)

## Description

Removes tag_profile from CapacityResource and uses swatch-product-configuration instead to validate granularity.

Also, removed RHEL productId from the api spec since it is no longer in use.

### Setup

1. Ensure account is opted in: 
```
curl -X 'PUT' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/opt-in' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```

### Steps

1. Run following curl command and ensure no errors:
```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/capacity/products/RHEL%20Server/Cores?granularity=Daily&beginning=2023-07-01T00%3A32%3A28Z&ending=2023-07-31T00%3A32%3A28Z' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```
1. Run following curl command and should get error that granularity is not supported for product:
```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/capacity/products/Satellite/Cores?granularity=Hourly&beginning=2023-07-01T00%3A32%3A28Z&ending=2023-07-31T00%3A32%3A28Z' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```

### CODE REVIEW

**Review by TrayvonMcKnight — APPROVED** [review]
https://drive.google.com/file/d/1xmoY8XFgfAq-xzOpPcHVM0zpZoZdbuxe/view?usp=drive_link

**Review by kahowell — APPROVED** [review]

**Comment by kahowell** [comment]
@kflahert FYI needs a rebase
@TrayvonMcKnight looks like there are a number of tests that use `productId: RHEL` that will need updates as a result of these changes.

**Comment by TrayvonMcKnight** [comment]
> @kflahert FYI needs a rebase @TrayvonMcKnight looks like there are a number of tests that use `productId: RHEL` that will need updates as a result of these changes.

Ack, I will spin up another ticket, following the new QE process

**Comment by ntkathole** [comment]
@kahowell @kflahert we are removing RHEL from api spec, but want to keep RHEL in tag_profile ?

**Comment by kflahert** [comment]
> @kahowell @kflahert we are removing RHEL from api spec, but want to keep RHEL in tag_profile ?

@ntkathole there is not entry for RHEL in the swatch-product-config.

File: api/rhsm-subscriptions-api-spec.yaml [hunk 1]
```diff
@@ -1140,7 +1140,6 @@ components:
       type: string
       enum:
         - OpenShift Container Platform
-        - RHEL
```
Comment (ntkathole) [inline api/rhsm-subscriptions-api-spec.yaml:1143]: How about changing examples of responses at other places from api/rhsm-subscriptions-api-spec.yaml ?
 
```
first: '/api/rhsm-subscriptions/v1/hosts/RHEL?sla=Premium&usage=Production&offset=0&limit=5'
last: '/api/rhsm-subscriptions/v1/hosts/RHEL?sla=Premium&usage=Production&offset=5&limit=5'
previous: null
next: '/api/rhsm-subscriptions/v1/hosts/RHEL?sla=Premium&usage=Production&offset=5&limit=5'
```

Comment (kflahert) [reply api/rhsm-subscriptions-api-spec.yaml:1143]: Yeah good catch will update!

**Pull request merged**